include/pinocchio/spatial/inertia.hpp

```diff
@@ -889,6 +960,366 @@ namespace pinocchio
+     * @param pseudo_inertia A 4x4 pseudo inertia matrix.
+     * @return A PseudoInertiaTpl object.
+     */
+    static PseudoInertiaTpl FromMatrix(const Matrix4 & pseudo_inertia)
+    {
+      Scalar mass = pseudo_inertia(3, 3);
+      Vector3 h = pseudo_inertia.template block<3, 1>(0, 3);
+      Matrix3 Sigma = pseudo_inertia.template block<3, 3>(0, 0);
+      return PseudoInertiaTpl(mass, h, Sigma);
+    }
+
+    /**
+     * @brief Constructs a PseudoInertiaTpl object from dynamic parameters.
+     * @param dynamic_params A 10-dimensional vector of dynamic parameters.
+     * @return A PseudoInertiaTpl object.
+     */
+    static PseudoInertiaTpl FromDynamicParameters(const Vector10 & dynamic_params)
+    {
+      Scalar mass = dynamic_params[0];
+      Vector3 h = dynamic_params.template segment<3>(1);
+      Matrix3 I_bar;
+      I_bar << dynamic_params[4], dynamic_params[5], dynamic_params[7], dynamic_params[5],
+        dynamic_params[6], dynamic_params[8], dynamic_params[7], dynamic_params[8],
+        dynamic_params[9];
+
+      Matrix3 Sigma = 0.5 * I_bar.trace() * Matrix3::Identity() - I_bar;
+      return PseudoInertiaTpl(mass, h, Sigma);
+    }
+
+    /**
+     * @brief Converts the PseudoInertiaTpl object to dynamic parameters.
+     * @return A 10-dimensional vector of dynamic parameters.
+     */
+    Vector10 toDynamicParameters() const
+    {
+      Matrix3 I_bar = Sigma.trace() * Matrix3::Identity() - Sigma;
+
+      Vector10 dynamic_params;
+      dynamic_params[0] = mass;
+      dynamic_params.template segment<3>(1) = h;
+      dynamic_params[4] = I_bar(0, 0);
+      dynamic_params[5] = I_bar(0, 1);
+      dynamic_params[6] = I_bar(1, 1);
+      dynamic_params[7] = I_bar(0, 2);
+      dynamic_params[8] = I_bar(1, 2);
+      dynamic_params[9] = I_bar(2, 2);
+
+      return dynamic_params;
+    }
+
+    /**
+     * @brief Constructs a PseudoInertiaTpl object from an InertiaTpl object.
+     * @param inertia An InertiaTpl object.
+     * @return A PseudoInertiaTpl object.
+     */
+    static PseudoInertiaTpl FromInertia(const InertiaTpl<Scalar, Options> & inertia)
+    {
+      Vector10 dynamic_params = inertia.toDynamicParameters();
+      return FromDynamicParameters(dynamic_params);
+    }
+
+    /**
+     * @brief Converts the PseudoInertiaTpl object to an InertiaTpl object.
+     * @return An InertiaTpl object.
+     */
+    InertiaTpl<Scalar, Options> toInertia() const
+    {
+      Vector10 dynamic_params = toDynamicParameters();
+      return InertiaTpl<Scalar, Options>::FromDynamicParameters(dynamic_params);
+    }
+
+    /**
+     * @brief Constructs a PseudoInertiaTpl object from log Cholesky parameters.
+     * @param log_cholesky A 10-dimensional vector of log Cholesky parameters.
+     * @return A PseudoInertiaTpl object.
+     */
+    static PseudoInertiaTpl
+    FromLogCholeskyParameters(const typename LogCholeskyParameters::Vector10 & log_cholesky)
```

Comment:
Like InetriaTpl, prefer initializing directly from LogCholeskyParameters. It's safer and you do the conversion just two line bellow.